scopes/harmony/application/deploy.task.ts

```diff
@@ -20,17 +19,20 @@ export class DeployTask implements BuildTask {
     const componentsResults = await mapSeries(apps, async (app): Promise<any> => {
       const aspectId = this.application.getAppAspect(app.name);
       if (!aspectId) return undefined;
-      const capsules = context.capsuleNetwork.seedersCapsules;
-      const capsule = this.getCapsule(capsules, aspectId);
-      if (!capsule?.component) return undefined;
+      const capsule = context.capsuleNetwork.seedersCapsules.getCapsuleIgnoreVersion(ComponentID.fromString(aspectId));
+      if (!capsule || !capsule?.component) return undefined;
       const buildTask = this.getBuildTask(context.previousTasksResults, context.envRuntime.id);
       if (!buildTask) return undefined;
-      const componentArtifacts = buildTask.artifacts?.get(capsule.component);
-      const artifactList = componentArtifacts?.[1];
-      if (!artifactList) return undefined;
-      const deployContext: DeployContext = Object.assign(context, { artifactList });
-      if (!capsule || !app.deploy) return undefined;
-      await app.deploy(deployContext, capsule);
+      const componentResults = buildTask.componentsResults.find((res) =>
+        res.component.id.isEqual(capsule.component.id, { ignoreVersion: true })
+      );
+      const metadata = componentResults?.metadata?.deployContext || {};
```

Comment:
See comment above on metadata

Reply:
we decide to pass the data on _metadata, and set ts-ignfore for that